src/routing.js

```diff
@@ -0,0 +1,76 @@
+import {chain, merge, toPairs, trim} from 'lodash';
+
+const isParam = segment => segment.substr(0, 1) === ':';
+const toSegments = pathname => trim(pathname, '/').split('/');
+
+const buildRoute = (segments, methods) => {
+  const segment = segments.shift();
+  const edge = segments.length === 0;
```

Comment:
I would rather use `isEdge` to indicate that this is a bool value.
`isLastSegment` is the better name here ( as you used below )